openedx/core/djangoapps/credentials/tests/test_utils.py

```diff
@@ -171,14 +190,37 @@ def test_get_programs_credentials(self):
         """
         # create credentials and program configuration
         self.create_credentials_config()
-        self.create_programs_config()
 
         # Mocking the API responses from programs and credentials
-        self.mock_programs_api()
-        self.mock_credentials_api(self.user, reset_url=False)
-        actual = get_programs_credentials(self.user)
-        expected = self.expected_credentials_display_data()
-
-        # Checking result is as expected
-        self.assertEqual(len(actual), 2)
-        self.assertEqual(actual, expected)
+        uuidfirst, uuidsecond = str(uuid.uuid4()), str(uuid.uuid4())
+        credentials_api_response = {
+            "next": None,
+            "results": [
+                factories.UserCredential(
+                    id=1, username='test',
+                    credential=factories.ProgramCredential(program_uuid=uuidfirst)
+                ),
+                factories.UserCredential(
+                    id=2, username='test',
+                    credential=factories.ProgramCredential(program_uuid=uuidsecond)
+                ),
+                factories.UserCredential(
+                    id=3, username='test',
+                    status='revoked',
+                    credential=factories.ProgramCredential()
+                )
+            ]
+        }
+        self.mock_credentials_api(self.user, data=credentials_api_response, reset_url=False)
+        programs = [
+            catalog_factories.Program(uuid=uuidfirst), catalog_factories.Program(uuid=uuidsecond)
+        ]
+
+        with mock.patch("openedx.core.djangoapps.catalog.utils.get_programs") as mock_get_programs:
```

Comment:
@MatthewPiatetsky don't we use responses in the catalog to mock external HTTP APIs? This is the same idea but with a different library.